docs/shortcuts.md

```diff
@@ -0,0 +1,154 @@
+
+### Request
+
+```http
+POST /shortcuts HTTP/1.1
+Host: alice.cozy.example
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": {
+    "type": "io.cozy.files.shortcuts",
+    "attributes": {
+      "name": "sunset.jpg.url",
+      "dir_id": "629fb233be550a21174ac8e19f003e4a",
+      "url": "https://alice-photos.cozy.example/#/photos/629fb233be550a21174ac8e19f0043af",
+      "metadata": {
+        "target": {
+          "cozyMetadata": {
+            "instance": "https://alice.cozy.example/"
+          },
+          "app": "photos",
+          "_type": "io.cozy.files",
+          "mime": "image/jpg"
+        }
+      }
+    }
+  }
+}
+```
+
+### Response
+
+```http
+HTTP/1.1 201 Created
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "_id": "629fb233be550a21174ac8e19f0043af",
+  "_rev": "1-61c7804bdb4f9f8dae5a363cb9a30dd8",
+  "type": "file",
+  "name": "sunset.jpg.url",
+  "dir_id": "629fb233be550a21174ac8e19f003e4a",
+  "trashed": false,
+  "md5sum": "vfEMDpJShs8QeIlsDmw9VA==",
+  "created_at": "2020-02-10T20:38:04Z",
+  "updated_at": "2020-02-10T20:38:04Z",
+  "tags": [],
+  "metadata": {
+    "target": {
+      "cozyMetadata": {
+        "instance": "https://alice.cozy.example/"
+      },
+      "app": "photos",
+      "_type": "io.cozy.files",
+      "mime": "image/jpg"
+    }
+  },
+  "size": 62,
+  "executable": false,
+  "class": "shortcut",
+  "mime": "application/shortcut",
+  "cozyMetadata": {
+    "doctypeVersion": 1,
+    "metadataVersion": 1,
+    "createdAt": "2020-02-10T20:38:04Z",
+    "createdOn": "https://bob.cozy.example/",
+    "updatedAt": "2020-02-10T20:38:04Z",
+    "uploadedAt": "2020-02-10T20:38:04Z",
+    "uploadedOn": "https://bob.cozy.example/"
+  }
+}
+```
+
+## GET /shortcuts/:id
+
+This route can be used to get information about a shortcut. A permission to
+read the file is required to use it. There are two variants, depending on the
+`Accept` header.
+
+### Request (JSON variant)
+
+If the `Accept` header is `application/vnd.api+json` (or `application/json`),
+the information are returned in JSON-API format.
+
+```http
+GET /shortcuts/629fb233be550a21174ac8e19f0043af HTTP/1.1
+Host: alice.cozy.example
+Accept: application/vnd.api+json
+```
+
+## Response (JSON variant)
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": {
+    "id": "629fb233be550a21174ac8e19f0043af",
+    "type": "io.cozy.files.shortcuts",
+    "attributes": {
+      "name": "sunset.jpg.url",
+      "dir_id": "629fb233be550a21174ac8e19f003e4a",
+      "url": "https://alice-photos.cozy.example/#/photos/629fb233be550a21174ac8e19f0043af",
+      "metadata": {
+        "target": {
+          "cozyMetadata": {
+            "instance": "https://alice.cozy.example/"
+          },
+          "app": "photos",
+          "_type": "io.cozy.files",
+          "mime": "image/jpg"
+        }
+      }
+    }
+  }
+}
+```
```

Comment:
Can you add an example with an external link? Do we receive the external "link" in the `data.attributes.url` attribute ?

Reply:
Yes, the `url` is filled by parsing the content of the file, and it will be here for an external link.